src/components/va-text-input/va-text-input.tsx

```diff
@@ -0,0 +1,146 @@
+import {
+  Component,
+  Element,
+  Prop,
+  Host,
+  h,
+  Event,
+  EventEmitter,
+} from '@stencil/core';
+
+@Component({
+  tag: 'va-text-input',
+  styleUrl: 'va-text-input.css',
+  shadow: true,
+})
+export class VaTextInput {
+  @Element() el: HTMLElement;
+
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  /**
+   * The label for the text input.
+   */
+  @Prop() label: string | HTMLElement;
+
+  /**
+   * The error message to render.
+   */
+  @Prop() error?: string | HTMLElement;
+
+  /**
+   * Set the input to required and render the (Required) text
+   */
+  @Prop() required?: boolean;
+
+  /**
+   * Placeholder text to show in the input field
+   */
+  @Prop() placeholder?: string;
+
+  /**
+   * The maximum number of characters allowed in the input.
+   */
+  @Prop() maxlength?: number;
+
+  /**
+   * What to tell the browser to auto-complete the field with.
+   */
+  @Prop() autocomplete?: string;
+
+  /**
+   * Don't emit any component-library-analytics events
+   */
+  @Prop() disableAnalytics?: boolean;
+
+  /*
+   * The value for the input.
+   */
+  @Prop({ mutable: true }) value?: string = '';
+  // TODO: Make the value prop reflective. Currently, it isn't because it screws
+  // up the input behavior. For now, the only bug is that the changed value
+  // isn't reflected in the DOM on the web component.
+  //
+  // $('va-text-input').value will be correct
+  // $('va-text-input').getAttribute('value') will be incorrect
+
+  private handleChange = (e: Event) => {
+    const target = e.target as HTMLInputElement;
+    this.value = target.value;
+  };
+
+  private handleBlur = () => {
+    if (this.disableAnalytics) return;
+
+    this.componentLibraryAnalytics.emit({
+      componentName: 'va-text-input',
```

Comment:
:thinking: I don't know what the google analytics side of this looks like, but I've been making `componentName` the same as the React implementations. Maybe we should get clarification on if this is automatic or if something has to be manually created for each new `componentName`.

Reply:
Ah, right. @bsmartin-ep Are you the one we talk to about this?

Reply:
I just merged this, but we can change the `componentName` here if we need to in another PR.

Reply:
Sorry @cvalarida and @bkjohnson - was out all week.

> Maybe we should get clarification on if this is automatic or if something has to be manually created for each new componentName.

Yes indeed.  New components need to be [added to this lookup table](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/site-wide/component-library-analytics-setup.js).  

Any objections to matching the react component name?  Is is inaccurate to use that for a web component?